src/tool/hpcrun/gpu/nvidia/cubin-symbols.c

```diff
@@ -187,19 +188,40 @@ relocateSymbolsHelper
   } else {
     return NULL;
   }
+
+  // If .symtab_shndx section exists, we need extended numbering
+  // for section index of a symbol
+  Elf_Data *symtab_shndx_data = NULL;
+  if (symtab_shndx_section != NULL) {
+    symtab_shndx_data = elf_getdata(symtab_shndx_section, NULL);
+  }
+
   Elf_Data *datap = elf_getdata(scn, NULL);
   if (datap) {
     symbol_values = newSymbolsVector(nsymbols);
     for (int i = 0; i < nsymbols; i++) {
       GElf_Sym sym;
-      GElf_Sym *symp = gelf_getsym(datap, i, &sym);
+      Elf32_Word xndx;
```

Comment:
The lookup method in the class would take (datap, i, &sym, unsigned long &index) and return symp.